bigquery_etl/cli/query.py

```diff
@@ -2118,148 +2120,52 @@ def deploy(
-                    )
-                    sys.exit(1)
+    _deploy = partial(
+        deploy_table,
+        destination_table=destination_table,
+        force=force,
+        use_cloud_function=use_cloud_function,
+        skip_existing=skip_existing,
+        respect_dryrun_skip=respect_dryrun_skip,
+        sql_dir=sql_dir,
+    )
 
-            bigquery_schema = existing_schema.to_bigquery_schema()
+    failed_deploys, skipped_deploys = [], []
+    with concurrent.futures.ThreadPoolExecutor(max_workers=parallelism) as executor:
+        future_to_query = {
+            executor.submit(_deploy, query_file): query_file
+            for query_file in query_files
+        }
+        for future in futures.as_completed(future_to_query):
+            query_file = future_to_query[future]
             try:
-                table = client.get_table(full_table_id)
-            except NotFound:
-                table = bigquery.Table(full_table_id)
-
-            table.schema = bigquery_schema
-            _attach_metadata(query_file_path, table)
-
-            if not table.created:
-                client.create_table(table, exists_ok=True)
-                click.echo(f"Destination table {full_table_id} created.")
-            elif not skip_existing:
-                client.update_table(
-                    table,
-                    [
-                        "schema",
-                        "friendly_name",
-                        "description",
-                        "time_partitioning",
-                        "clustering_fields",
-                        "labels",
-                    ],
-                )
-                click.echo(f"Schema (and metadata) updated for {full_table_id}.")
-        except Exception:
-            print_exc()
-            return query_file
-
-    with ThreadPool(parallelism) as pool:
-        failed_deploys = [r for r in pool.map(_deploy, query_files) if r]
+                future.result()
+            except SkippedDeployException as e:
+                print(f"Skipped deploy for {query_file}: ({e})")
+                skipped_deploys.append(query_file)
+            except FailedDeployException as e:
+                print(f"Failed deploy for {query_file}: ({e})")
+                failed_deploys.append(query_file)
+            else:
+                print(f"{query_file} successfully deployed!")
 
     if not skip_external_data:
         failed_external_deploys = _deploy_external_data(
             name, sql_dir, project_id, skip_existing
         )
         failed_deploys += failed_external_deploys
 
-    if len(failed_deploys) > 0:
+    if skipped_deploys:
+        click.echo("The following deploys were skipped:")
+        for skipped_deploy in skipped_deploys:
+            click.echo(skipped_deploy)
+
+    if failed_deploys:
         click.echo("The following tables could not be deployed:")
         for failed_deploy in failed_deploys:
             click.echo(failed_deploy)
         sys.exit(1)
```

Comment:
It'd be nice if this could collect and raise the various exceptions as an `ExceptionGroup`.